src/promptflow-tracing/promptflow/tracing/_span_wrapper.py

```diff
@@ -0,0 +1,71 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+
+class SpanWrapper:
+    """
+    A wrapper class for spans.
+    """
+
+    def __init__(self, wrapped: object):
+        """
+        Initialize the SpanWrapper with the object to be wrapped.
+
+        :param wrapped: The object to be wrapped.
+        """
+        object.__setattr__(self, "__wrapped__", wrapped)
+        self._self_should_end = True
+
+    def __getattr__(self, name: str) -> object:
+        """
+        Get attribute from the wrapped object.
+
+        :param name: The name of the attribute.
+        :return: The attribute of the wrapped object.
+        """
+        if name == "__wrapped__":
```

Comment:
Looks hard to understand this part, please add more clear explanation.

Reply:
Removed to use private attribute on `Span` itself instead.